[Chen Yanyu] iP

YaleChen299 opened the pull request:
# Hiko (A version of Duke created by Yanyu)

> Eye can zoo eat! 
> -A. Tan ([source](https://www.youtube.com/watch?v=dQw4w9WgXcQ))

Hiko frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

YaleChen299 committed:
Complete Level-1

YaleChen299 committed:
Complete Level-2

YaleChen299 committed:
Complete Level-3

YaleChen299 committed:
Complete Level-4

YaleChen299 committed:
Complete A-TextUiTesting

YaleChen299 committed:
Complete Level-5

YaleChen299 committed:
Complete Level-6

YaleChen299 committed:
A-Enums

YaleChen299 committed:
fix code style

YaleChen299 committed:
Add Save functionality

YaleChen299 committed:
fix code style

YaleChen299 committed:
fix directory file bugs

YaleChen299 committed:
support for datetime

YaleChen299 committed:
Merge branch 'branch-Level-7' into master

YaleChen299 committed:
Merge branch 'branch-Level-8' into master

YaleChen299 committed:
refactor to more oop

YaleChen299 committed:
Complete A-package

YaleChen299 committed:
refactor Task and add Junit Test

YaleChen299 committed:
Add jar configuration

YaleChen299 committed:
Add JavaDoc

YaleChen299 committed:
Update coding stadard

YaleChen299 committed:
add find feature

YaleChen299 committed:
Merge branch 'branch-A-JavaDoc' into master

YaleChen299 committed:
Merge branch 'branch-A-CodingStandard' into master

YaleChen299 committed:
Merge branch 'branch-Level-9' into master

SkyBlaise99 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+
+    /**
+     * This will overwrite the file at the specified path with the provided tasks.
+     *
+     * @param tasks a List of task to overwrite to the file
+     * @throws IOException thrown during writing
+     */
+    public void write(List<Task> tasks) throws IOException {
+        Files.createDirectories(Paths.get("data"));
+        file.createNewFile();
+        FileWriter fw = new FileWriter(filePath);
+        StringBuilder text = new StringBuilder();
+        for (Task task : tasks) {
+            String taskString = String.join(deliminator, task.saveStrings()) + "\n";
+            text.append(taskString);
+        }
+        fw.write(text.toString());
+        fw.close();
```
Perhaps you can add some line breaks for better readability?

YaleChen299 replied:
Thanks for the comment! I have added `checkstyle` and fixed some of the issues.

SkyBlaise99 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,42 @@
+
+    public static void showGreeting() {
+        String logo =
+                " _   _ _ _\n" +
+                        "| | | (_) | _____\n" +
+                        "| |_| | | |/ / _ \\\n" +
+                        "|  _  | |   < (_) |\n" +
+                        "|_| |_|_|_|\\_\\___/\n";
```
Perhaps shifting `+` to the front of the line?
``` java
    String logo = "..."
            + "..."
            + "...";
```

SkyBlaise99 reviewed (commented):
Well done! I can't really find any major code quality violations. I believe that once you start working on `A-checkstyle`, it will show you all the small little errors and warnings that I can't possibly spot manually, and ask you to correct them. Keep up with the great work 👍

YaleChen299 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into master

YaleChen299 committed:
change main class name

YaleChen299 committed:
add checkstyle and fix style error

e0543403 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    /**
+     * return the save string arrays for file writing
```
Shouldn't this be Returns instead of return? Also the period at the end is missing I think.

YaleChen299 replied:
Thanks for the suggestion! I will fix soon!

e0543403 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    /**
+     * return the save string arrays for file writing
```
Shouldn't this be Returns instead of return? Also the period at the end is missing I think.

e0543403 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,67 @@
+
+/**
+ * This is the parent for all Exceptions that can be thrown
+ * when interacting with a Duke
```
Isn't there supposed to be a period at the end?

e0543403 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * An Event is a Task that recognises /at time
```
Isn't there supposed to be a period at the end?

e0543403 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+    }
+
+    /**
+     * load the file and parse into a list of task.
```
Should load be Loads instead?

e0543403 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+    }
+
+    /**
+     * This will overwrite the file at the specified path with the provided tasks.
```
Perhaps this should be Overwrites the file at the specified path with the provided tasks.

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * return the string representation of the task status.
```
Should return be Returns?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * return the string representation of the task status.
+     *
+     * @return X if the task is done
```
Should there be a period at the end?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * return the description.
```
Should return be Returns?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * return the description.
+     *
+     * @return the description of the task
```
Should the be The and there be a period at the end?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * return the status of the task.
```
Should return be Returns?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * return the status of the task.
+     *
+     * @return the status of the task
```
Should the be The and there be a period at the end?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * Mark the task as done.
```
Should Mark be Marks?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * return the save string arrays for file writing
```
Should return be Returns and there be  a period at the end?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * return the save string arrays for file writing
+     *
+     * @return the array of Strings representing the task
```
Should the be The and there be a period at the end?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    /**
+     * save the task list to the local path.
```
Should save be Saves?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    /**
+     * remove the task from the list.
```
Should remove be Removes?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+     * remove the task from the list.
+     *
+     * @param index the index starting with 1.
+     * @throws IndexOutOfBoundsException thrown when index out of bound
```
Should the first letter of a statement be capitalized and there be a period at the end?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+     * save the task list to the local path.
+     *
+     * @param storage the Storage instance
+     * @throws IOException thrown when there is an error saving
```
Should the first letter in statement be capitalized and there be a period at the end?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+package duke;
+
+/**
+ * A representation of task with description and status
```
Should there be a period at the end?

e0543403 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+     * This will overwrite the file at the specified path with the provided tasks.
+     *
+     * @param tasks a List of task to overwrite to the file
+     * @throws IOException thrown during writing
```
Should the first letter of a statement be capitalized and there be a period at the end?

e0543403 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,130 @@
+     *
+     * @param input input string from the user.
+     * @return the command instance.
+     * @throws DukeException thrown when there are error parsing.
```
Should the first letter of a statement should be capitalized?

e0543403 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+     *
+     * @return the list of task for display
+     * @throws FileNotFoundException thrown when a file is not found
+     * @throws FileFormatException   thrown when the file is wrongly formatted
```
Should the first letter in a line be capitalized and a period be at the end of a statement?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    /**
+     * add task to the list.
```
Should add be Adds?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+    /**
+     * add task to the list.
+     *
+     * @param task the task to be added.
```
Should the be The?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    /**
+     * mark the task as done.
```
Should mark be Marks?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+    /**
+     * mark the task as done.
+     *
+     * @param index the index starting with 1.
```
Should the be The?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    /**
+     * filter out the tasks by the search string and print the result.
```
Should filter be Filters?

e0543403 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,112 @@
+    /**
+     * filter out the tasks by the search string and print the result.
+     *
+     * @param searchString the string contained by the description.
```
Should the be The?

e0543403 commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,28 @@
+package duke;
+
+/**
+ * simplest kind of task
```
Should simplest be Simplest and there be a period at the end?

e0543403 commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,28 @@
+    /**
+     * return the save string arrays for file writing
+     *
+     * @return the array of Strings representing the task
```
Should the first letter of the statement be capitalized and a period at the end of the line?

e0543403 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    /**
+     * This method is called at the start of the program to greet the user.
```
Perhaps it should be Greets the user when called at the start of the program?

e0543403 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,145 @@
+     * @param tasks   the TaskList object that is used to interact with the tasks.
+     * @param ui      the ui object used to interact with the user
+     * @param storage the storage object used to read/write to files.
+     * @throws DukeException any exception thrown when interacting with Duke.
```
Should the first letter in a statement be capitalized and a period at the end of the ui statement?

e0543403 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,145 @@
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
+
+    /**
+     * getter for the Action.
```
Should be Gets the Action?

e0543403 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,145 @@
+    /**
+     * getter for the Action.
+     *
+     * @return the String representing action
```
Should the be The and there be a period at the end of the line?

e0543403 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    /**
+     * getter for the Argument.
```
Should be Gets the Argument?

e0543403 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,145 @@
+    /**
+     * getter for the Argument.
+     *
+     * @return the String representing the argument
```
Should the be The and there be a period at the end of the line?

e0543403 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    /**
+     * @return the true if the action is BYE;
```
Should the be The?

e0543403 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A deadline is a Task that recognises /by deadline
```
Should there be a period at the end of the statement?

e0543403 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+    /**
+     * return the save string arrays for file writing
+     *
+     * @return the array of Strings representing the task
```
Should the be The and there be a period at the end of the statement?

e0543403 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+    private TaskList tasks;
+
+    /**
+     * public constructor for Duke.
```
Should be Constructs a Hiko chatbot?

e0543403 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+    /**
+     * public constructor for Duke.
+     *
+     * @param path the path for the saved file.
```
Should the be The?

e0543403 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+    /**
+     * This is the public constructor for storage.
+     *
+     * @param filePath the path in which the file is stored.
```
Should the be The?

e0543403 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+    private String status;
+
+    /**
+     * This is the public constructor for storage.
```
Should be Constructs a storage with data writing and loading functionality?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    private boolean isDone;
+
+    /**
+     * Public Constructor for Task.
```
Should be Constructs a Task?

e0543403 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * Public Constructor for Task.
+     *
+     * @param description the description for the task.
```
Should the be The?

e0543403 commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    /**
+     * return the save string arrays for file writing
```
Should return be Returns?

e0543403 reviewed (commented):
A large number of Javadoc nits, most of which were similar and would not reduce readability by any amount, was not able to find any coding standard violations in the code. Overall keep up the good work.

YaleChen299 committed:
add gui

YaleChen299 committed:
Merge branch 'branch-Level-10' into master

YaleChen299 committed:
fix styling

YaleChen299 committed:
Fix coding style

This commit fixes various coding style and improve code quality.

Some of the changes include renaming variables and adding java doc.

YaleChen299 committed:
add assertions

YaleChen299 committed:
Merge pull request #3 from YaleChen299/branch-A-CodeQuality

Fix coding style

YaleChen299 committed:
Merge pull request #4 from YaleChen299/branch-A-Assertions

add assertions

YaleChen299 committed:
Add Stream

YaleChen299 committed:
Create gradle.yml

YaleChen299 committed:
Fix style

YaleChen299 committed:
Fix style

YaleChen299 committed:
FIx test

YaleChen299 committed:
Add Help

YaleChen299 committed:
Improve Gui Responsiveness

YaleChen299 committed:
improve user experience

YaleChen299 committed:
Add user input history feature

YaleChen299 committed:
refactors code, adds clear, improves ui

YaleChen299 committed:
Add files via upload

YaleChen299 committed:
Set theme jekyll-theme-slate

YaleChen299 committed:
rename

YaleChen299 committed:
Merge branch 'master' of https://github.com/YaleChen299/ip into master

YaleChen299 committed:
Update table of content

YaleChen299 committed:
Set theme jekyll-theme-cayman

YaleChen299 committed:
Update user guide

YaleChen299 committed:
fix styling

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md

YaleChen299 committed:
Update README.md
